File: test/com/jetbrains/android/dsl/tests/functional/LayoutsTest.java
package com.jetbrains.android.dsl.tests.functional;

import com.jetbrains.android.dsl.BaseGeneratorProps;
import com.jetbrains.android.dsl.Subsystem;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

public class LayoutsTest extends BaseFunctionalTest {
    private final String testDataFile = "com/jetbrains/android/dsl/tests/functional/LayoutsTest.kt";

    @Override
    @BeforeMethod
    public void setUp() throws Exception {
        super.setUp();
    }

    protected void initSettings(BaseGeneratorProps settings) {
        settings.setGenerateLayoutParamsHelperClasses(true);
    }

    @Test
    public void testLayouts() throws Exception {
        runFunctionalTest(testDataFile, Subsystem.LAYOUTS);
    }
}

File: test/org/jetbrains/android/dsl/functional/LayoutsTest.kt
package org.jetbrains.android.dsl.functional
import org.jetbrains.android.dsl.*
import org.testng.annotations.Test
public class LayoutsTest : AbstractFunctionalTest() {
    private val testDataFile = "LayoutsTest.kt"
    override fun initSettings(settings: BaseGeneratorConfiguration) {
        settings.files.add(KoanFile.LAYOUTS)
    }
    [Test] public fun testLayoutsTestFor21() {
        runFunctionalTest(testDataFile, KoanFile.LAYOUTS, "21")
    }
    [Test] public fun testLayoutsTestFor21s() {
        runFunctionalTest(testDataFile, KoanFile.LAYOUTS, "21s")
    }
}
